sdk/provisioning/Azure.Provisioning/src/resourcemanager/ResourceGroup.cs

```diff
@@ -16,6 +16,17 @@ public class ResourceGroup : Resource<ResourceGroupData>
     {
         internal static readonly ResourceType ResourceType = "Microsoft.Resources/resourceGroups";
 
+        /// <summary>
+        /// Whether the resource group is anonymous.
+        /// </summary>
+        internal bool IsAnonymous { get; }
+
+        internal ResourceGroup(IConstruct scope, string? name = "rg", string version = "2023-07-01", AzureLocation? location = default, bool isAnonymous = false, Subscription? parent = default)
```

Comment:
instead of adding `isAnonymous` can we have the callers pass in `resourceGroup()` for the name?  All callers should be internal right?